doc/run_experiment.rst

```diff
@@ -984,6 +984,69 @@ most probable class for each instance. Only really makes a difference
 when storing predictions. Defaults to ``False``. Note that this also
 applies to the tuning objective.
 
+.. _pipeline:
+
+pipeline *(Optional)*
+"""""""""""""""""""""
+
+Whether or not the final learner object should contain a ``pipeline``
+attribute that contains a scikit-learn `Pipeline <http://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html>`__ object composed
+of copies of each of the following steps of training the learner:
+
+    * feature vectorization (`vectorizer`)
+    * feature selection (`selector`)
+    * feature sampling (`sampler`) 
+    * feature scaling (`scaler`)
+    * main estimator (`estimator`)
+
+The strings in the parentheses represent the name given to each
+step in the pipeline. 
+
+The goal of this attribute is to allow better interoperability 
+between SKLL learner objects and scikit-learn. The user can 
+train the model in SKLL and then further tweak or analyze 
+the pipeline in scikit-learn, if needed. Each component of the 
+pipeline is a (deep) copy of the component that was fit as part 
+of the SKLL model training process. We use copies since we do 
+not want the  original SKLL model to be affected if the user 
+modifies the components of the pipeline in scikit-learn space.
+
+Here's an example of how to use this attribute.
+
+.. code-block:: python
+
+    from sklearn.preprocessing import LabelEncoder
+
+    from skll import Learner
+    from skll.data import CSVReader, NDJReader
+
+    # train a classifier and a regressor using the SKLL API
+    fs1 = NDJReader('examples/iris/train/example_iris_features.jsonlines').read()
+    learner1 = Learner('LogisticRegression', pipeline=True)
+    learner1.train(fs1, grid_search=True, grid_objective='f1_score_macro')
+
+    fs2 = CSVReader('examples/boston/train/example_boston_features.csv').read()
+    learner2 = Learner('RescaledSVR', feature_scaling='both', pipeline=True)
+    learner2.train(fs2, grid_search=True, grid_objective='pearson')
+    
+    # now, we can explore the stored pipelines in sklearn space
+    enc = LabelEncoder().fit(fs1.labels)
+
+    # first, the classifier
+    D1 = {"f0": 6.1, "f1": 2.8, "f2": 4.7, "f3": 1.2}
+    pipeline1 = learner1.pipeline
+    enc.inverse_transform(pipeline1.predict(D1))
```

Comment:
I noticed that, for the second example, I was able to predict on the pipeline directly without issue, but when I tried to reverse engineer it given what was in the pipeline, the issue with densification came up. I'm wondering, why doesn't that show up in the pipeline?
```python
In [136]: list(learner2.pipeline.named_steps)
Out[136]: ['vectorizer', 'selector', 'scaler', 'estimator']
```

Reply:
Ok, I see why now (I believe):
```python
In [140]: learner2.pipeline.named_steps['vectorizer'].sparse
Out[140]: False

In [141]: learner2.feat_vectorizer.sparse
Out[141]: True
```

Reply:
I wonder, though, if this should be done because it might cause confusion if the vectorizer in the pipeline and that at the top level are different. Can't we just add the `Densifier` step and not set `sparse` to `False` instead?

Reply:
I preferred to optimized the trade-off the other way: it's probably better not to insert _any_ non-sklearn classes especially since it's pretty hacky class to begin with. Therefore, I only add `Densifier` when it absolutely necessary (using `FeatureHasher` or `SkewedChi2Sampler`).  

Perhaps we can add a warning saying that `sparse` is being turned off? Thoughts?